### ISSUE TITLE
Nested "traced", or tracing an Action, instead of IO?

### ISSUE THREAD

**Issue by saurabhnanda** [ndmitchell/shake#775]
Due to the type of `traced` one cannot use it in a nested manner, or trace an `Action` (instead of an `IO`). Is this intended to specifically restrict what can, and cannot, be traced?

**Comment by ndmitchell** [comment]
Yes. The idea is that this way traced operations are non-overlapping, and you can assign costs to traced operations easily. There are certainly ways that traced could have been made more powerful and assigned costs in other ways, but this is a simple one. Traced really serves two masters - printing to the screen and profile reports.

As I think about it now, it actually wouldn't have been too hard to make it nest properly...
